src/azul/plugins/repository/dss/__init__.py

```diff
@@ -387,17 +401,14 @@ class DSSFileDownload(RepositoryFileDownload):
 
     def update(self, plugin: RepositoryPlugin) -> None:
         self.drs_path = None  # to shorten the retry URLs
-        dss_endpoint = config.dss_endpoint
-        url = dss_endpoint + '/files/' + urllib.parse.quote(self.file_uuid, safe='')
         if self.replica is None:
             self.replica = 'aws'
-        dss_params = {
-            'version': self.file_version,
-            'replica': self.replica
-        }
+        url = plugin.direct_file_url(self.file_uuid, self.file_version, self.replica)
```

Comment:
> We shouldn't pass an optional kw arg as a positional, and we should start enforceing kw-only args wherever possible.

I'm not certain what you mean "optional kw arg". You made two distinct changes, adding a default value for `file_version` and making `file_version` and `replica` keyword-only. So are you referring to:

1. All arguments with type `Optional`?
2. All arguments with default values?
3. All arguments satisfying 1 or 2?

Reply:
Neither. I'm referring to the `replica` parameter. You declared it as a keyword argument but passed it as a positional. One motivation for keyword arguments is to allow for future modification of a function's signature like adding keyword arguments or  reordering them, without having to modify every call site. With your calling style, a keyword argument added before `replica` would unintentionally consume the third passed argument. You could of course argue now that I used inaccurate terminology in my comment, but I would like to encourage you to focus your energy at your own contributions rather than litigating my feedback.

Reply:
Resolved at standup today.